src/AutoMapper/QueryableExtensions/ProjectionExpression.cs

```diff
@@ -104,7 +105,10 @@ internal class MemberVisitor : ExpressionVisitor
         {
             protected override Expression VisitLambda<T>(Expression<T> node)
             {
-                var memberExpression = node.Body as MemberExpression;
+                MemberExpression memberExpression = 
+                    node.Body.NodeType == ExpressionType.Convert 
+                        ? (node.Body as UnaryExpression)?.Operand  as MemberExpression
+                        :  node.Body as MemberExpression;
```

Comment:
That's not what I meant, but I guess it doesn't matter. There is a simpler approach. 
The test should look like [this](https://github.com/AutoMapper/AutoMapper/blob/master/src/IntegrationTests.Net4/ByteArray.cs). An in memory test is not relevant because EF supports only certain expressions and that's what matters.